### PR TITLE
Add support for conv2d explicit quantization

### PR DESCRIPTION
**Pull request by leo0519**
### PR types

New features

### PR changes

Others

### Description

This PR adds conv2d support for explicit quantization.

### CODE REVIEW

**Comment by leo0519** [comment]
Rebase to the latest develop branch and fix the code conflict.

**Comment by leo0519** [comment]
Fix some CI failures (add timeout and pre-commit).

**Comment by leo0519** [comment]
CI status
- PR-CI-APPROVAL: 
```sh
2023-09-25 23:43:43 0. Unittest is not allowed to be disabled.
2023-09-25 23:43:43 You must have one RD (kolinwei(Recommend), wanghuancoder, luotao1, QingshuChen, qili93 or ZzSean or Aurelius84) approval for the usage of @unittest.skip or @unittest.skipIf.
2023-09-25 23:43:43 +@unittest.skipIf(
2023-09-25 23:43:43 +@unittest.skipIf(
2023-09-25 23:43:43 +@unittest.skipIf(
2023-09-25 23:43:43 +@unittest.skipIf(
2023-09-25 23:43:43 1. You must have one QA (XieYunshen(Recommend) or chalsliu) approval for setting parameter RUN_TYPE as EXCLUSIVE, DIST, NIGHTLY, EXCLUSIVE:NIGHTLY or DISTNIGHTLY, or setting parameter SERIAL, or setting TIMEOUT properties.
2023-09-25 23:43:43 The corresponding lines are as follows:
2023-09-25 23:43:43 test/ir/inference/CMakeLists.txt
2023-09-25 23:43:43 +  set_tests_properties(test_trt_explicit_quantization_resnet PROPERTIES TIMEOUT
2023-09-25 23:43:43 +                       PROPERTIES TIMEOUT 300)
2023-09-25 23:43:43 For more information, please refer to:https://github.com/PaddlePaddle/Paddle/wiki/PaddlePaddle-Unit-test-specification
2023-09-25 23:43:43 There are 2 approved errors.
```
- PR-CI-Coverage: The CI environment is TRT 7.X, but the explicit quantization is supported with TRT >= 8. Hence, this CI failure should be exempted.

File: test/ir/inference/test_trt_explicit_quantization.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+    def test_model(self):
+        self.build_program()
+        baseline = self.inference(precision_mode=PrecisionType.Float32)
+        predict = self.inference(precision_mode=PrecisionType.Int8)
+        np.testing.assert_allclose(predict, baseline, rtol=1e-2, atol=1e-2)
+
+
+@unittest.skipIf(
+    paddle.inference.get_trt_compile_version() < (8, 5, 1),
```
Comment (Wangzheee) [inline test/ir/inference/test_trt_explicit_quantization.py:74]: 单测中跳过了TenosorRT8.5，显示量化的代码中有跳过吗？

Comment (leo0519) [reply test/ir/inference/test_trt_explicit_quantization.py:74]: 雖然 TRT 8.0 便支持顯式量化，但在 8.5 時有改變 quant axis 設置的方式，設置方式如下
- per-tensor: 不需要設置 axis (默認)
- per-channel: 需設置 axis 範圍 [-r, r-1] (r = rank)，axis -r = axis 0, axis -r+1 = axis 1, ...

Comment (leo0519) [reply test/ir/inference/test_trt_explicit_quantization.py:74]: 新的 [commit](https://github.com/PaddlePaddle/Paddle/pull/57160/commits/81fde9cc48b00bf314319f4617d969a022789021) 中更新了 TRT 版本的檢查

**Review by Wangzheee — APPROVED** [review]
LGTM

**Review by qili93 — APPROVED** [review]
LGTM for unittest.skipIf

**Review by XieYunshen — APPROVED** [review]
LGTM for `  set_tests_properties(test_trt_explicit_quantization_resnet PROPERTIES TIMEOUT
                                                                        300)
  set_tests_properties(test_trt_explicit_quantization_mobilenet
                       PROPERTIES TIMEOUT 300)`

**Pull request merged**